### PR TITLE
Add getters for aws_client() and remove all usages of maps:get(<something>, Client)

### PR DESCRIPTION
**Pull request by onno-vos-dev**
Will follow up with a PR to [aws-codegen](https://github.com/aws-beam/aws-codegen) to fix the `maps:get(..., Client)` from the autogenerated code but this one would require merging prior to that. :+1:

### CODE REVIEW

**Review by philss — APPROVED** [review]

**Review by jadeallenx — APPROVED** [review]

File: src/aws_client.erl [hunk 1]
```diff
@@ -8,13 +8,25 @@
...
 -type token() :: binary().
 -type http_port() :: binary().
--type aws_client() :: map().
+-type proto() :: binary().
+-type service() :: binary().
 -type endpoint() :: binary().
+-type aws_client() :: map().
 
 -export_type([access_key_id/0, secret_access_key/0, region/0,
               token/0, aws_client/0]).
```
Comment (andreashasse) [inline src/aws_client.erl:32]: What happens if you make aws_client/0 opaque?

File: src/aws_client.erl [hunk 2]
```diff
@@ -97,6 +109,40 @@ make_local_client(AccessKeyID, SecretAccessKey, Port, Endpoint)
...
+  AccessKeyId.
+
+-spec secret_access_key(aws_client()) -> secret_access_key().
+secret_access_key(#{secret_access_key := SecretAccessKey} = _Client) ->
+  SecretAccessKey.
+
+-spec token(aws_client()) -> token().
+token(#{token := Token} = _Client) ->
+  Token;
+token(_) ->
```
Comment (andreashasse) [inline src/aws_client.erl:123]: Maybe match that is a map here.

**Review by andreashasse — APPROVED** [review]

File: src/aws_client.erl [hunk 3]
```diff
@@ -26,7 +26,7 @@
 -type proto() :: binary().
 -type service() :: binary().
 -type endpoint() :: binary().
--type aws_client() :: map().
+-opaque aws_client() :: map().
 
 -export_type([access_key_id/0, secret_access_key/0, region/0,
               token/0, aws_client/0]).
```
Comment (andreashasse) [inline src/aws_client.erl:32]: Aren't opaque types atomically exported?

**Review by andreashasse — APPROVED** [review]

File: src/aws_client.erl (hunk 3 above)
Comment (onno-vos-dev) [reply src/aws_client.erl:32]: I get a warning if I remove the export: `opaque type aws_client() is not exported`

**Pull request merged**